linux/power_operation_scripts/set_power_script.yml

```diff
@@ -115,26 +115,54 @@
 - name: "Check the log file"
   when: power_script_op != "suspend"
   block:
-    - name: "Search the log file {{ script_file_path }}"
-      ansible.builtin.shell: "grep {{ script_file_path }} *.log"
-      args:
-        chdir: "/var/log/vmware-tools"
-      delegate_to: "{{ vm_guest_ip }}"
-      register: result_search_str
-      until:
-        - result_search_str.rc is defined
-        - result_search_str.rc == 0
-      retries: 30
-      delay: 5
-      ignore_errors: true
-
-    - name: "Check the serching result"
-      ansible.builtin.assert:
-        that:
-          - result_search_str.rc is defined
-          - result_search_str.rc == 0
-        fail_msg: "Failed to find the {{ script_file_path }} in {{ vmtools_vmsvc_log_file }} for {{ power_script_type }} {{ power_script_op }}."
-        success_msg: "Successfully found the {{ script_file_path }} in {{ vmtools_vmsvc_log_file }} for {{ power_script_type }} {{ power_script_op }}."
+    - name: "Check the log file in {{ guest_os_ansible_distribution }}"
+      when: guest_os_ansible_distribution != 'FreeBSD'
+      block:
+        - name: "Search the log file {{ script_file_path }} in {{ guest_os_ansible_distribution }}"
+          ansible.builtin.shell: "grep {{ script_file_path }} *.log"
+          args:
+            chdir: "/var/log/vmware-tools"
+          delegate_to: "{{ vm_guest_ip }}"
+          register: result_search_str
+          until:
+            - result_search_str.rc is defined
+            - result_search_str.rc == 0
+          retries: 30
+          delay: 5
+          ignore_errors: true
+
+        - name: "Check the serching result"
+          ansible.builtin.assert:
+            that:
+              - result_search_str.rc is defined
+              - result_search_str.rc == 0
+            fail_msg: "Failed to find the {{ script_file_path }} in {{ vmtools_vmsvc_log_file }} for {{ power_script_type }} {{ power_script_op }}."
+            success_msg: "Successfully found the {{ script_file_path }} in {{ vmtools_vmsvc_log_file }} for {{ power_script_type }} {{ power_script_op }}."
+   
+    - name: "Check the log file in {{ guest_os_ansible_distribution }}"
+      when: guest_os_ansible_distribution == 'FreeBSD'
+      block:
+        - name: "Search the log file {{ script_file_path }} in {{ guest_os_ansible_distribution }}"
+          ansible.builtin.shell: "grep {{ script_file_path }} *vmsvc*.log ./vmware-tools/*.log"
+          args:
+            chdir: "/var/log"
```

Comment:
Why does FreeBSD have a different log directory when tools.conf has below content:
```
[logging]
log=true
vmtoolsd.level=debug
vmtoolsd.handler=file
vmtoolsd.data=/var/log/vmware-tools/vmtoolsd.log
vmsvc.level=debug
vmsvc.handler=file
vmsvc.data=/var/log/vmware-tools/vmsvc.log
```

Reply:
Change to only check ../vmware-tools/.log. Testing is on-going...